File: app/src/main/java/com/example/jetpackmvvm/sample/util/MainViewModel.java
package com.example.jetpackmvvm.sample.util;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import org.jetbrains.annotations.NotNull;

public class MainViewModel extends AndroidViewModel {

    private final MutableLiveData<Boolean> showBanner = new MutableLiveData<Boolean>(true);

    public MainViewModel(@NonNull @NotNull Application application) {
        super(application);
    }

    public void setBannerState(Boolean showBanner) {
        this.showBanner.setValue(showBanner);
    }
    public LiveData<Boolean> getBannerState() {
        return showBanner;
    }
}

File: app/src/main/java/com/terry/jetpackmvvm/sample/util/MainViewModel.kt
package com.terry.jetpackmvvm.sample.util

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class MainViewModel : ViewModel() {
    private val _showBanner = MutableLiveData(true)

    val showBanner: LiveData<Boolean>
        get() = _showBanner

    fun setBannerState(isShow: Boolean) {
        _showBanner.postValue(isShow)
    }
}